pkg/scraper/scraper.go

```diff
@@ -228,6 +229,7 @@ func (c *Scraper) getNodeInfo(node *corev1.Node) (NodeInfo, error) {
 	info := NodeInfo{
 		Name:           node.Name,
 		ConnectAddress: addr,
+		KubeletPort:    node.Status.DaemonEndpoints.KubeletEndpoint.Port,
```

Comment:
How do you plan to distinquish between `read-only` and `secure` port?

Reply:
Is this field expected to be not empty on all k8s versions 1.8+ ? 
If not we should add information that 0.4.0 has changed supported versions in README https://github.com/kubernetes-sigs/metrics-server/blob/master/README.md#deployment

Can this value be empty (or any field in between)? Could we fallback to previous default or `kubelet-port` flag?

Reply:
KubeletEndpoint mainly records the communication port between kubelet and apiserver, which is secure port. I think it should not be empty in theory. I tested several versions of k8s, such as 1.14 and 1.17. There is no problem. I think using the previous method as the default way to set the port, and using this as an additional option to adapt to other needs is a reasonal way.

Reply:
About possibility of empty `node.Status.DaemonEndpoints.KubeletEndpoint.Port`, I think it's reasonable to assume that status can be empty during node bootstrap. I don't think we should skip validation of data. 
Can you at least check for `nil` to avoid potential panic? We can just skip this node and write a log.

Reply:
One clarification (not 100% sure), but it's Kubelet that initiates connection to API server. This connection is required for correct Node behaviour. Reverse connection (apiserver->kubelet) is not crucial in every k8s cluster. I know it's used for `kubectl exec`, `kubectl port-forward`,

Reply:
Thx for your explanation. It really make senses.